sound/soc/sof/intel/hda-dai.c

```diff
@@ -419,11 +440,19 @@ static int ssp_dai_hw_params(struct snd_pcm_substream *substream,
 
 		if (!strcmp(dai->name, sof_dai->cpu_dai_name) &&
 		    substream->stream == sof_dai->comp_dai.direction) {
-			config = &sof_dai->dai_config[sof_dai->current_config];
-			if (!config) {
-				dev_err(sdev->dev, "error: no config for DAI %s\n", sof_dai->name);
+			if (sof_dai->current_config > sof_dai->number_configs - 1) {
+				dev_err(sdev->dev, "error: DAI config index out of range\n");
 				return -EINVAL;
 			}
+			/* we can return immediately as there are no choises */
+			if (sof_dai->number_configs < 2)
+				config = &sof_dai->dai_config[sof_dai->current_config];
+			/* try to match to pcm params */
+			if (!config)
+				config = ssp_dai_config_pcm_params_match(sdev, sof_dai, params);
```

Comment:
This is interesting. @plbossart @ranj063 @lyakh  We discussed with Jaska and basicly this matching part can be done here (in DAI/BE hw_params), but it could also be done in sof/pcm.c:sof_pcm_dai_link_fixup().

It would seem from ASoC point of view both would be good.

Order-wise fixup is called first, so it would actually be cleanest solution to have the matching logic there, in sof/pcm.c. 

But if fixup is meant primarily to adjust for information coming from topology, then the matching part does not belong there.

Any insights on which is the better place?

Reply:
Yes, should be done in sof_pcm_dai_link_fixup()

Reply:
ok, moving there